[REVIEW] Update gpuCI documentation with how to handle auto-merger failures (#31)
rapidsai/docs: maintainers/gpuci.md

Each branch is built in a container using CUDA 10.0 and Python 3.6. the purpose of this job is primarily for testing with a GPU present.

#### Auto-mergers

> During the release process, the branch for the next release will be created and set as default. Once this happens, the auto-merger branch jobs will be activated. Auto-mergers will automatically merge any commits made to the release branch to the latest default branch during burn down.

Reviewer: ```suggestion
During the release process, the branch for the next release is created and set as default. Once this happens, the auto-merger branch jobs are activated. Auto-mergers automatically merge any commits made to the release branch to the latest default branch during burn down.
```
